src/CSET/operators/read.py

```diff
@@ -429,6 +431,64 @@ def _fix_pressure_coord_callback(cube: iris.cube.Cube):
                 cube.coord("pressure").convert_units("hPa")
 
 
+def _lfric_time_callback(cube: iris.cube.Cube):
+    """
+    Fix time coordinate metadata if missing dimensions.
+
+    Some model data does not contain forecast_reference_time or forecast_period as
+    expected coordinates, and so we cannot aggregate over case studies without this
+    metadata. This callback fixes these issues.
+
+    Notes
+    -----
+    Some parts of the code have been adapted from Paul Earnshaw's scripts.
+    """
+    # Construct forecast_reference time and forecast_period if they dont exist.
+    if not cube.coords("forecast_reference_time"):
+        try:
+            tcoord = cube.coord("time")
+            init_time = datetime.datetime.fromisoformat(
+                cube.coord("time").attributes["time_origin"]
+            )
+            time_unit = cube.coord("time").units
+            frt_point = time_unit.date2num(init_time)
+
+            frt_coord = iris.coords.AuxCoord(
+                frt_point, units=tcoord.units, standard_name="forecast_reference_time"
+            )
+            cube.add_aux_coord(frt_coord)
+        except KeyError:
+            logging.info(
+                "Cannot find forecast_reference_time, but no time_origin to construct it"
+            )
+
+    # Construct forecast_period axis (leadtime) if doesn't exist.
+    if not cube.coords("forecast_period"):
+        try:
+            tcoord = cube.coord("time")
+            init = cube.coord("forecast_reference_time")
+
+            lead_times = tcoord.points - init.points
+            if "seconds" in str(tcoord.units):
+                units = "seconds"
+            elif "hours" in str(tcoord.units):
+                units = "hours"
+            else:
+                raise ValueError(f"Not a recognised base time unit {str(tcoord.units)}")
```

Comment:
```suggestion
                raise ValueError(f"Not a recognised base time unit {tcoord.units}")
```
F-strings automatically call `str` on arguments.